lib/relations/Relation.js

```diff
@@ -70,12 +70,14 @@ class Relation {
     }
   }
 
-  joinModelClass(knex) {
-    if (knex && knex !== this.joinTableModelClass.knex()) {
-      return this.joinTableModelClass.bindKnex(knex);
-    } else {
-      return this.joinTableModelClass;
+  get joinModelClass() {
+    if (this.joinTableModelClass) {
+      const knex = this.ownerModelClass.knex();
+      if (knex !== this.joinTableModelClass.knex()) {
+        return this.joinTableModelClass.bindKnex(knex);
```

Comment:
@koskimas couldn't this simply be `this.joinTableModelClass.knex(knex)` now instead?